Add setting for controlling MailChimp reader default status

claudiulodro opened the pull request:
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- Describe the changes made to this Pull Request, and the reason for such changes. -->

Closes [an issue we've been encountering on a couple client sites](https://app.asana.com/0/1206477424334739/1206714953071858/f).

The original reasoning behind making the default status 'transactional' in MailChimp when people register on a site is correct at a strictly technical level because they technically haven't signed up to receive mailings from the site, but have merely registered on it. However **in practice** it is not the behavior publishers expect and it prevents them from doing e.g. a welcome series of emails when someone registers on the site.

This PR adds a new setting for controlling the default status of these readers in MailChimp. 

### How to test the changes in this Pull Request:

1. In the Engagement wizard, make sure you have this reader activity data setting set with an audience ID as well as some newsletters available in your Registration modal:

<img width="1065" alt="Screenshot 2024-03-07 at 2 26 52 PM" src="https://github.com/Automattic/newspack-plugin/assets/7317227/971ddba7-de52-4702-88cc-988ecdbcb9f6">

2. You should now see a new setting **if an audience is set** for controlling the default status. Leave it in "Transactional/Non-Subscribed" for now.

<img width="883" alt="Screenshot 2024-03-15 at 10 14 00 AM" src="https://github.com/Automattic/newspack-plugin/assets/7317227/2ed5e6db-b122-409d-a5b7-3c66fc694f52">

4. In an incognito window, register on the site but **do not** select any newsletters:

<img width="824" alt="Screenshot 2024-03-15 at 10 13 02 AM" src="https://github.com/Automattic/newspack-plugin/assets/7317227/94a49854-9aec-4a9d-bfcd-28cbb1d5d95b">

5. In MailChimp, observe the reader is in "Non-Subscribed" status:

<img width="362" alt="Screenshot 2024-03-15 at 10 13 32 AM" src="https://github.com/Automattic/newspack-plugin/assets/7317227/55a1e1f0-7e18-43bc-ba82-bdccf06f5a2f">

6. Set the setting to "Subscribed"

<img width="873" alt="Screenshot 2024-03-15 at 10 14 08 AM" src="https://github.com/Automattic/newspack-plugin/assets/7317227/297c0b73-9e46-4cbd-bee2-811631a55982">

7. In an incognito window, register on the site again and **do not** select any newsletters:

<img width="801" alt="Screenshot 2024-03-15 at 10 19 02 AM" src="https://github.com/Automattic/newspack-plugin/assets/7317227/5351bc0f-22d0-45fe-bdce-53483ec36cf7">

8. In MailChimp, observe the reader is in "Subscribed" status:

<img width="360" alt="Screenshot 2024-03-15 at 10 19 24 AM" src="https://github.com/Automattic/newspack-plugin/assets/7317227/2584b1b9-1c5b-4f30-9c76-cd6f009cbb79">

9. Just to be safe, set the setting back to "Transactional/Non-Subscribed" and register on the site but **do** select a newsletter. In MailChimp the user should be in "Subscribed" status:

<img width="800" alt="Screenshot 2024-03-15 at 10 20 11 AM" src="https://github.com/Automattic/newspack-plugin/assets/7317227/92c24512-611f-48a7-85f5-e6b23782e082">
<img width="349" alt="Screenshot 2024-03-15 at 10 20 35 AM" src="https://github.com/Automattic/newspack-plugin/assets/7317227/b9024ff8-ca5d-44d2-967a-f919444f867c">


### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

<!-- Mark completed items with an [x] -->

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1206740961392205

adekbadek committed:
fix: allow media lib access if explicitly set

Adjusts the overly restrictive changes from https://github.com/Automattic/newspack-plugin/pull/1541

adekbadek committed:
fix(media-partners): handle skip-in-feeds when exporting to apple news

adekbadek committed:
fix(newsletters): handle status metadata

dependabot[bot] committed:
chore(deps): bump google/auth from 1.35.0 to 1.37.0

Bumps [google/auth](https://github.com/googleapis/google-auth-library-php) from 1.35.0 to 1.37.0.
- [Release notes](https://github.com/googleapis/google-auth-library-php/releases)
- [Changelog](https://github.com/googleapis/google-auth-library-php/blob/main/CHANGELOG.md)
- [Commits](https://github.com/googleapis/google-auth-library-php/compare/v1.35.0...v1.37.0)

---
updated-dependencies:
- dependency-name: google/auth
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
chore(deps-dev): bump phpunit/phpunit from 9.6.16 to 9.6.17

Bumps [phpunit/phpunit](https://github.com/sebastianbergmann/phpunit) from 9.6.16 to 9.6.17.
- [Changelog](https://github.com/sebastianbergmann/phpunit/blob/9.6.17/ChangeLog-9.6.md)
- [Commits](https://github.com/sebastianbergmann/phpunit/compare/9.6.16...9.6.17)

---
updated-dependencies:
- dependency-name: phpunit/phpunit
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
chore(deps-dev): bump @wordpress/browserslist-config

Bumps [@wordpress/browserslist-config](https://github.com/WordPress/gutenberg/tree/HEAD/packages/browserslist-config) from 5.34.0 to 5.35.0.
- [Release notes](https://github.com/WordPress/gutenberg/releases)
- [Changelog](https://github.com/WordPress/gutenberg/blob/trunk/packages/browserslist-config/CHANGELOG.md)
- [Commits](https://github.com/WordPress/gutenberg/commits/@wordpress/browserslist-config@5.35.0/packages/browserslist-config)

---
updated-dependencies:
- dependency-name: "@wordpress/browserslist-config"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

chickenn00dle committed:
fix(GA): update spelling of referrer param (#2917)

This adds another `referrer` param (in addition to `referer`) to GA4 events to account for the expected double-r spelling.

laurelfulford committed:
fix: update text domains for the analytics, components (#2953)

laurelfulford committed:
fix: update text domains for the advertising wizard (#2952)

laurelfulford committed:
fix: update text domain in wizards/connections directory (#2956)

laurelfulford committed:
fix: correct text domains in wizards/engagement (#2957)

jaredrethman committed:
feat(webhooks): add `newspack_webhooks_process_request_errors` action (#2955)

dependabot[bot] committed:
chore(deps): bump @babel/preset-env from 7.23.9 to 7.24.0

Bumps [@babel/preset-env](https://github.com/babel/babel/tree/HEAD/packages/babel-preset-env) from 7.23.9 to 7.24.0.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.24.0/packages/babel-preset-env)

---
updated-dependencies:
- dependency-name: "@babel/preset-env"
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
chore(deps-dev): bump @types/qs from 6.9.11 to 6.9.12

Bumps [@types/qs](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/qs) from 6.9.11 to 6.9.12.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/qs)

---
updated-dependencies:
- dependency-name: "@types/qs"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
chore(deps): bump @babel/plugin-transform-runtime from 7.23.9 to 7.24.0

Bumps [@babel/plugin-transform-runtime](https://github.com/babel/babel/tree/HEAD/packages/babel-plugin-transform-runtime) from 7.23.9 to 7.24.0.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.24.0/packages/babel-plugin-transform-runtime)

---
updated-dependencies:
- dependency-name: "@babel/plugin-transform-runtime"
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

chickenn00dle committed:
fix(ras): check for auth status before updating my account label (#2970)

This PR addresses an issue where the RAS my account button label was incorrectly outputting the logged in state label after a logged in user has logged out.

adekbadek committed:
fix(image-credits): handle errors when fetching metadata (#2968)

Co-authored-by: Jared Rethman <jaredrethman@gmail.com>

jaredrethman committed:
fix: handle no themes are available (#2967)

laurelfulford committed:
fix: correct text domains for handoff, healthcheck wizards (#2961)

laurelfulford committed:
fix: correct text domains for popup wizard (#2962)

jaredrethman committed:
feat: jetpack default modules (#2959)

* feat(jetpack): added default active modules

* fix: indent

* refactor: updated prop visibility

* refactor: callback name

* refactor: jp module activation on newspack setup complete

adekbadek committed:
feat: add a filter to esp normalized contact (#2940)

cab8925d5fb550c4f7e71347c06c575971924c70

adekbadek committed:
fix: data-events tweaks (#2935)

f5bdc5b7dedd8ccb36b5b06515fa66ea1f0c2476

adekbadek committed:
feat(core-image-block): display media credit in editor; handle distributed posts (#2936)

f9c165a4f6a0cfd73c3c4ee9c911a178f71a6c8b

chickenn00dle committed:
fix(ras): reset otp inputs when switching login actions (#2971)

This addresses an issue where logging in with OTP does not reset OTP inputs when a reader attempts to either have another code sent OR sign in with a different email address

claudiulodro committed:
fix: default subscribed status in data events

dkoo commented:
(Updated title so that when merging it will trigger a hotfix release)

dkoo reviewed (approved):
If almost all publishers are sending newsletter content by group/tag/segment or some other means of sub-grouping their audience contacts, then this is the correct behavior. I think we decided on `transactional` originally because we can't guarantee that a publisher won't have their MC account set up to send content to the entire audience, and we can't enforce it either way. But I'm okay with this as long as it's the behavior publishers expect.

claudiulodro commented:
Would it be "safer" if I made it a checkbox/toggle in that reader activity data section? Or would that muddle the UI? @Automattic/newspack-design what do you think?

dkoo committed:
chore: namespace WP functions in plugin-manager class (#2978)

dkoo committed:
fix: allow + character in reader account email (#2980)

dependabot[bot] committed:
chore(deps-dev): bump @types/wordpress__block-editor

Bumps [@types/wordpress__block-editor](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/wordpress__block-editor) from 11.5.10 to 11.5.11.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/wordpress__block-editor)

---
updated-dependencies:
- dependency-name: "@types/wordpress__block-editor"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
chore(deps): bump qs from 6.11.2 to 6.12.0

Bumps [qs](https://github.com/ljharb/qs) from 6.11.2 to 6.12.0.
- [Changelog](https://github.com/ljharb/qs/blob/main/CHANGELOG.md)
- [Commits](https://github.com/ljharb/qs/compare/v6.11.2...v6.12.0)

---
updated-dependencies:
- dependency-name: qs
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
chore(deps-dev): bump @wordpress/browserslist-config

Bumps [@wordpress/browserslist-config](https://github.com/WordPress/gutenberg/tree/HEAD/packages/browserslist-config) from 5.35.0 to 5.36.0.
- [Release notes](https://github.com/WordPress/gutenberg/releases)
- [Changelog](https://github.com/WordPress/gutenberg/blob/trunk/packages/browserslist-config/CHANGELOG.md)
- [Commits](https://github.com/WordPress/gutenberg/commits/@wordpress/browserslist-config@5.36.0/packages/browserslist-config)

---
updated-dependencies:
- dependency-name: "@wordpress/browserslist-config"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

laurelfulford committed:
fix: add a check for setup complete before adding homepage pattern (#2987)

semantic-release-bot committed:
chore(release): 3.3.3 [skip ci]

## [3.3.3](https://github.com/Automattic/newspack-plugin/compare/v3.3.2...v3.3.3) (2024-03-12)

### Bug Fixes

* add a check for setup complete before adding homepage pattern ([#2987](https://github.com/Automattic/newspack-plugin/issues/2987)) ([36238e9](https://github.com/Automattic/newspack-plugin/commit/36238e96a863723720979520400a9afc65dd0cfa))

chickenn00dle committed:
feat: add ID to ad unit dropdown labels (#2988)

This PR adds the ad unit id to the placement settings ad unit dropdown to help publishers more easily identify ad units.

chickenn00dle committed:
fix: render correct gam onboarding flow and update service account support link (#2986)

This PR fixes an issue where the advertising wizard was always rendering the service account onboarding flow even when the oauth flow should have been available. It also updates a google support link to a more relevant page.

claudiulodro committed:
feat: add setting for MC default reader status

claudiulodro committed:
fix: js linter is finicky

claudiulodro commented:
I've updated this PR with a setting, changed it from a hotfix (since this is now a new feature), and revised the testing instructions. Can I please get another review? Thanks!

dependabot[bot] committed:
chore(deps-dev): bump @types/wordpress__block-editor

Bumps [@types/wordpress__block-editor](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/wordpress__block-editor) from 11.5.11 to 11.5.12.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/wordpress__block-editor)

---
updated-dependencies:
- dependency-name: "@types/wordpress__block-editor"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dkoo committed:
refactor: move status set outside loop

dkoo committed:
fix: fix fallback logic in get_order_data() util method

adekbadek committed:
fix(memberships-gate): handle no comments (#2999)

semantic-release-bot committed:
chore(release): 3.3.4 [skip ci]

## [3.3.4](https://github.com/Automattic/newspack-plugin/compare/v3.3.3...v3.3.4) (2024-03-20)

### Bug Fixes

* **memberships-gate:** handle no comments ([#2999](https://github.com/Automattic/newspack-plugin/issues/2999)) ([7973c8e](https://github.com/Automattic/newspack-plugin/commit/7973c8ea85faea4fcfb50a69a9ad8d03324a09f9))

dkoo commented on includes/data-events/connectors/class-mailchimp.php:
```diff
@@ -62,6 +62,20 @@ private static function get_audience_id() {
+			'subscribed',
+		];
+		$default_status = Reader_Activation::get_setting( 'mailchimp_reader_default_status' );
+		return in_array( $default_status, $allowed_statuses ) ? $default_status : 'transactional';
```
```suggestion
		return in_array( $default_status, $allowed_statuses, true ) ? $default_status : 'transactional';
```

CI might throw a warning if we don't use `in_array` with strict type checking

dkoo reviewed (approved):
Working as described! left one suggestion, but since it's so minor we can just merge this as soon as that's accepted.

dkoo committed:
Merge branch 'trunk' into fix/mcstatus

dkoo committed:
Merge branch 'trunk' into fix/mcstatus
